include/uart.h

```diff
@@ -64,6 +65,60 @@ extern "C" {
  */
 #define UART_ERROR_BREAK    (1 << 3)
 
+/**
+ * @brief UART controller configuration structure
+ *
+ * @param baudrate   Baudrate setting in bps.
+ * @param flags      Bit field with following parts:
+ *
+ *     parity       [ 0 : 2 ]    - Parity bits.
+ *     stop_bits    [ 3 : 4 ]    - Stop bits.
+ *     data_bits    [ 5 : 7 ]    - Data bits.
+ *     flow_ctrl    [ 8 : 9 ]    - Flow control setting.
+ */
+struct uart_config {
+	u32_t baudrate;
+	u32_t flags;
+};
```

Comment:
Can we add a vendor, or optional?

Reply:
As in other APIs we try not to expose any vendor specific stuff, the problem is that code using this API would not be guaranteed to behave the same (or work at all) depending on such configuration.

Reply:
I'm against adding vendor config, if more options are needed they should be explicitly added as new configuration option

Reply:
@galak @pfalcon @nordic-krch Any comments on this? Other discussions seem to be resolved

Reply:
Let's leave US-timezone folks time to catch up/reply. I'd be ready to +1 this tomorrow.

P.S. There's a proposed PR process: https://docs.google.com/document/d/1dSLSTckOgCeBh8MrU4AgpjkR6hQ0IikPnreAOO_wWGY/edit